lib/model/model.go

```diff
@@ -2510,15 +2510,35 @@ func (m *model) Revert(folder string) {
 	runner.Revert()
 }
 
-func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly bool) map[string]interface{} {
+type TreeEntry struct {
+	Name     string                `json:"name"`
+	ModTime  time.Time             `json:"modTime"`
+	Size     int64                 `json:"size"`
+	Type     protocol.FileInfoType `json:"type"`
+	Children []*TreeEntry          `json:"children,omitempty"`
```

Comment:
Why list and not a map?

Reply:
Because maps suck for REST APIs.

Hypothetically, if you tried to consume this from let's say C, you'd get extra pain because of maps.